push-to-s3-and-invalidate-cloudfront/action.yml

```
@@ -67,60 +76,65 @@ runs:
       run: |
         if ${{ inputs.s3-dir-to-upload-to }} == "pr-previews"
         then
-          dir-to-upload-to=${{ inputs.s3-dir-to-upload-to }}/PR-${{ inputs.pull-request-number }}
+          dir_to_upload_to=${{ inputs.s3-dir-to-upload-to }}/PR-${{ inputs.pull-request-number }}
         else if ${{ inputs.s3-dir-to-upload-to }} == "/"
-          dir-to-upload-to=""
+          dir_to_upload_to=""
         else
-          dir-to-upload-to=${{ inputs.s3-dir-to-upload-to }}
+          dir_to_upload_to=${{ inputs.s3-dir-to-upload-to }}
         fi
-        echo $dir-to-upload-to >> $GITHUB_OUTPUT
+        echo "dir_to_upload_to=$dir_to_upload_to" >> $GITHUB_OUTPUT
 
     - name: Compute the S3 directory to delete from
       shell: bash
       id: delete
       run: |
         if ${{ inputs.s3-dir-to-delete-from }} == "pr-previews"
         then
-          dir-to-delete-from=${{ inputs.s3-dir-to-delete-from }}/PR-${{ inputs.pull-request-number }}
-        else if ${{ inputs.s3-s3-dir-to-delete-from }} == "/"
-          dir-to-delete-from=""
+          dir_to_delete_from=${{ inputs.s3-dir-to-delete-from }}/PR-${{ inputs.pull-request-number }}
+        else if ${{ inputs.s3-dir-to-delete-from }} == "/"
+          dir_to_delete_from=""
         else
-          dir-to-delete-from=${{ inputs.s3-dir-to-delete-from }}
+          dir_to_delete_from=${{ inputs.s3-dir-to-delete-from }}
+        fi
+        echo "dir_to_delete_from=$dir_to_delete_from" >> $GITHUB_OUTPUT
+
+    - name: Compute files to exclude
+      shell: bash
+      id: exclude
+      run: |
+        files_to_exlude=""
+        if [[ ${{ inputs.s3-files-to-exclude }} == "pr-previews" ]]
+        then
+          files_to_exlude="--exclude \"${{ inputs.s3-files-to-exclude }}\""
+        else if [[ ${{ inputs.s3-files-to-exclude }} == *,* ]]
+        then
+          IFS=','
+          for i in `echo "${{ inputs.s3-files-to-exclude }}"`
+          do
+            files_to_exlude="$files_to_exlude --exclude \"$i\""
+          done
+          unset IFS
         fi
-        echo $dir-to-delete-from >> $GITHUB_OUTPUT
+        echo "files_to_exlude=$files_to_exlude" >> $GITHUB_OUTPUT
 
-        
     - name: Upload the files to S3
       shell: bash
-      if: inputs.s3-action.lowercase == 'upload' || (inputs.s3-action == "" && (github.event_name == 'pull_request' && contains(fromJson('["opened", "synchronize"]'), github.event.action)))
+      if: inputs.s3-action.lowercase == 'upload' || (inputs.s3-action == "" && github.event_name == 'pull_request' && (contains(fromJson('["opened", "synchronize"]'), github.event.action) == 'true'))

```

```suggestion
      if: inputs.s3-action.lowercase == 'upload' || (inputs.s3-action == '' && github.event_name == 'pull_request' && (contains(fromJson('["opened", "synchronize"]'), github.event.action) == 'true')) || (inputs.s3-action == '' && github.event_name == 'push')
```
Small adjustment since we'd want this to trigger on-push as well

---

I was planning to trigger this on push to specific branches but that will make this complex for no reason. For now the push logic would be handled by the workflow calling this action